packages/cli-plugin-deploy-pulumi/commands/deploy.js

```diff
@@ -82,6 +82,17 @@ module.exports = async (inputs, context) => {
         pulumi: pulumi
     });
 
+    if (stack.app) {
```

Comment:
Nope. Under the hood Automation API uses pulumi CLI as a child process, so basically, that's just a more of a wrapper, than a completely different approach.
Context switching is because we have different pulumi login for every app, so running `getStackOutput` runs `pulumi login` and changes current folder for state files.

Reply:
OK